web-server/assets/src/script/skills.js

```diff
@@ -76,7 +76,24 @@ const filterSkillsList = filter => {
   })
 }
 
+const hideEmptyFilters = (filterElements) => {
+  filterElements.each((index, filter) => {
+    const filters = ($(filter).text() || '').trim().toLowerCase().split(/\s+/)
```

Comment:
this looks like its a duplicate of like 69. Instead of copying this logic lets move it. 

move the logic of splitting a search string into its parts from like 69 and into `matchFilters` and rename it `matchFilter` and have it to the part splitting.